### PR TITLE
[encoder] fix zero len array decoding

### PR DESCRIPTION
**Pull request by playboyko**
- strings.Split with empty slice argument returns slice with 1 element
which is empty string. without check len decoder crashes on empty
numeric arrays
- encoder test added
- types test added

### CODE REVIEW

File: encoder.go [hunk 1]
```diff
@@ -136,6 +136,10 @@ func (d *textDecoder) Decode(t string, value []byte) (driver.Value, error) {
 	if strings.HasPrefix(t, "Array") {
 		if len(v) > 0 && v[0] == '[' && v[len(value)-1] == ']' {
 			items := strings.Split(v[1:len(value)-1], ",")
```
Comment (bgaifullin) [inline encoder.go]: Thank you for PR.
I think the following construction will be better:
```
if len(v) > 0 && v[0] == '[' && v[len(v)-1] == '] {
    var items []string
    // check that array is not empty ([])
    if len(v) > 2 {
        items = strings.Split(v[1:len(v)-1], ",")
    }
```

**Comment by bgaifullin** [comment]
Thank you. the patch looks good to me.

Could you also stash your commits, by using command `git rebase -i HEAD^^` ?
because the second commit it is just some fix for first commit.

**Comment by playboyko** [comment]
@bgaifullin Could you please check if everything is ok now?

**Comment by bgaifullin** [comment]
Thanks.

**Comment by bgaifullin** [comment]
merged.

**Pull request merged**